### PR TITLE
chore: bump cairo to 0.10.0

### PR DESCRIPTION
**Pull request by tserg**
### What I did

Bump Cairo to 0.10.0
- Drop Python 3.7 and 3.8
- Add validation checks for Account contract
- `cairo-migrate` test contracts

### How I did it

As above.

### How to verify it

Tests pass.

### Checklist

- [X] Passes all linting checks (pre-commit and CI jobs)
- [ ] New test cases have been added and are passing
- [X] Documentation has been updated
- [X] PR title follows [Conventional Commit](https://www.conventionalcommits.org/en/v1.0.0/) standard (will be automatically included in the changelog)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -4,7 +4,7 @@ Ape compiler plugin around [the Cairo language](https://github.com/starkware-lib
 
 ## Dependencies
 
-* [python3](https://www.python.org/downloads) version 3.7.2 or greater, python3-dev
+* [python3](https://www.python.org/downloads) version 3.9.0 or greater, python3-dev
```
Comment (unparalleled-js) [inline README.md:7]: I believe `3.8` is the min

```suggestion
* [python3](https://www.python.org/downloads) version 3.8 or greater, python3-dev
```

File: pyproject.toml [hunk 2]
```diff
@@ -14,7 +14,7 @@ write_to = "ape_cairo/version.py"
 
 [tool.black]
 line-length = 100
-target-version = ['py37', 'py38', 'py39']
+target-version = ['py39']
```
Comment (unparalleled-js) [inline pyproject.toml:17]:
```suggestion
target-version = ['py38', 'py39']
```

File: setup.py [hunk 3]
```diff
@@ -53,13 +53,12 @@
...
-        "cairo-lang>=0.9.1,<0.10",
-        "starknet.py>=0.4.1a0,<0.5",
+        "cairo-lang>=0.10.0,<0.11",
+        "starknet.py>=0.5.1a0,<0.6",
         "eth-ape>=0.4.0,<0.5.0",
         "ethpm-types",
-        "importlib-metadata ; python_version<'3.8'",
     ],
-    python_requires=">=3.7.2,<4",
+    python_requires=">=3.9.0,<4",
```
Comment (unparalleled-js) [inline setup.py:61]:
```suggestion
    python_requires=">=3.8,<4",
```

File: setup.py [hunk 4]
```diff
@@ -75,8 +74,6 @@
         "Operating System :: MacOS",
         "Operating System :: POSIX",
         "Programming Language :: Python :: 3",
-        "Programming Language :: Python :: 3.7",
-        "Programming Language :: Python :: 3.8",
```
Comment (unparalleled-js) [inline setup.py:79]: keep?

File: tests/contracts/oz_proxy_lib.cairo [hunk 5]
```diff
@@ -1,117 +1,102 @@
-# SPDX-License-Identifier: MIT
```
Comment (unparalleled-js) [inline tests/contracts/oz_proxy_lib.cairo:1]: I have always started this work and I wanted to make the `openzeppelin` in structure closer in resemblance - I push that change here if that it is ok

**Review by unparalleled-js — COMMENTED** [review]
standby! Thanks!!

File: tests/contracts/oz_proxy_lib.cairo (hunk 5 above)
Comment (tserg) [reply tests/contracts/oz_proxy_lib.cairo:1]: of course!

**Comment by tserg** [comment]
1. Should python 3.7 be dropped from the CI? 
2. The default python version is 3.8 in the CI. Should it be changed to 3.9 following Cairo 0.10.0?

**Comment by unparalleled-js** [comment]
> 1. Should python 3.7 be dropped from the CI?

Yes!

> 2. The default python version is 3.8 in the CI. Should it be changed to 3.9 following Cairo 0.10.0?

We can use 3.9 to match the rest of the ecosystem. However - 3.8 is still supported.

**Pull request merged**